### PR TITLE
expose AnchorTitled component

### PR DESCRIPTION
**Pull request by TalZa1810**
Expose `<AnchorTitled/>` component so it can be used within the cheatsheet

### CODE REVIEW

File: packages/wix-storybook-utils/src/AnchoredTitle/index.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    };
+
+    const isChildren = React.Children.count(children) > 0;
+
+    const onAnchorClicked = event => isChildren? onCopy(event): event.preventDefault();
+
+    return (
+        <div
+            className={styles.titleContainer}
+            onClick={!isChildren ? onCopy : null}
```
Comment (TalZa1810) [inline packages/wix-storybook-utils/src/AnchoredTitle/index.tsx:45]: this was changed

File: packages/wix-storybook-utils/src/AnchoredTitle/index.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                data-tip
+                className={styles.link}
+                id={id}
+                href={`#${id}`}
+                target="_self"
+                onClick={onAnchorClicked}
+            >
+                <LinkSmall />
+            </a>
+            { isChildren ?
```
Comment (TalZa1810) [inline packages/wix-storybook-utils/src/AnchoredTitle/index.tsx:61]: this was changed

File: packages/wix-storybook-utils/src/AnchoredTitle/index.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    const onMouseEnter = () => ReactTooltip.show(copyLinkRef.current);
+
+    const onMouseLeave = () => {
+        ReactTooltip.hide(copyLinkRef.current);
+        setCopied(false);
+    };
+
+    const isChildren = React.Children.count(children) > 0;
+
+    const onAnchorClicked = event => isChildren? onCopy(event): event.preventDefault();
```
Comment (TalZa1810) [inline packages/wix-storybook-utils/src/AnchoredTitle/index.tsx:40]: adding a check here `isChildren`

**Comment by TalZa1810** [comment]
Hey @mykas,
I've added an example to the components section within storybook.
I wasn't sure whether I should add unit tests for this component.
Also, do I need to update types definitions? I saw this file but wasn't sure where should I update the types:
`packages/wix-storybook-utils/src/typings/story-section.ts`.


CC: @milkyfruit 
I just wanted to make sure, UX wise, that it's ok to use this component within the cheatsheet so that users can share a link to a specific example within the cheatsheet as well like we have with regular storybook examples.

File: packages/wix-storybook-utils/src/AnchoredTitle/index.tsx (hunk 1 above)
Comment (mykas) [reply packages/wix-storybook-utils/src/AnchoredTitle/index.tsx:45]:
```suggestion
            onClick={!isChildren && onCopy}
```

**Review by mykas — CHANGES_REQUESTED** [review]

File: packages/wix-storybook-utils/src/AnchoredTitle/index.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import Markdown from '../Markdown';
+import styles from './styles.scss';
+import sectionWithSiblingsStyles from '../Sections/section-with-siblings/styles.scss'
+
+const translations = {
+    copyLink: 'Copy Link',
+    linkCopiedToClipboard: 'Copied',
+};
+
+export const AnchoredTitle = ({ title, children }) => {
```
Comment (mykas) [inline packages/wix-storybook-utils/src/AnchoredTitle/index.tsx:16]: I think you missed to update the place where this AnchoredTitle is used in Section folder.

Comment (mykas) [reply packages/wix-storybook-utils/src/AnchoredTitle/index.tsx:16]: https://github.com/wix/wix-ui/blob/ef8346bb7aecef383e7ddd8582d5f44bc185a13b/packages/wix-storybook-utils/src/Sections/section-with-siblings/index.tsx#L11

**Comment by mykas** [comment]
> packages/wix-storybook-utils/src/typings/

Typings there are for storybook sections so there is no need for that. Regarding unit tests this component did not have any, but it would be great if you could add some knowing that we have now another place that uses it.

**Review by mykas — APPROVED** [review]

**Pull request merged**